tgui/packages/tgui/interfaces/MinimapController.js

```diff
@@ -0,0 +1,245 @@
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, ByondUi, Dropdown, Flex, Image, Input, Modal, NumberInput, Section, Stack } from '../components';
+import { Window } from '../layouts';
+import { capitalize } from './common/stringUtils';
+
+export const MinimapController = (params, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    title,
+    theme,
+    minimap_id,
+    markers_visible,
+    selecting_coordinates,
+    minimap_markers,
+    showNewMarkerMenu,
+    placable_marker_states,
+    placable_marker_images,
+    icon,
+    image,
+    x,
+    y,
+  } = data;
+
+  const [name, setName] = useLocalState(context, 'name');
+
+  const toggleNewMarkerMenu = () => {
+    data.showNewMarkerMenu = !showNewMarkerMenu;
```

Comment:
Updating values on the `data` object feels a bit weird to me. If this is intended to be local state, use local state. If this is intended to persist across users, use backend state (and `act`) like you did elsewhere.